### PR TITLE
Set up Prettier, hooks & format consistently

### PR DESCRIPTION
**Pull request by infotexture**
Per https://github.com/infotexture/dita-bootstrap.lunr/pull/9#discussion_r1242046458

### CODE REVIEW

**Comment by infotexture** [comment]
@jason-fox To minimize diffs with upstream files, I've excluded the Lunr JavaScript from formatting here:

https://github.com/infotexture/dita-bootstrap.lunr/blob/feature/checks/.prettierignore#L1-L4

If there's anything else that's copied in from elsewhere, we can add those files to the ignore list.

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# This file specifies files that Prettier should not format
+
+# Skip Lunr JavaScript files
+resource/**/*.js
```
Comment (jason-fox) [inline .prettierignore:4]: The only lunr.js file is `resource/lunr.js` - the others are mine.

```suggestion
resource/lunr.js
```

Should probably add this to `.gitattributes` too:

```
resource/lunr.js linguist-vendored
```

**Comment by jason-fox** [comment]
`resource/languages` should be excluded too - the stemmers and stop words are from a different library.

**Comment by infotexture** [comment]
> `resource/languages` should be excluded too - the stemmers and stop words are from a different library.

OK, thought so. Will amend and force-push to update.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "prepare": "husky install",
+    "prettier": "prettier --write \"**/*\" --ignore-unknown",
+    "prettier-check": "prettier --check \"**/*\" --ignore-unknown",
+    "test": "prettier --list-different \"**/*\" --ignore-unknown"
+  },
+  "lint-staged": {
+    "**/*": "prettier --write --ignore-unknown"
+  },
+  "devDependencies": {
+    "@prettier/plugin-xml": "3.1.0",
```
Comment (jason-fox) [inline package.json:24]: Add https://www.npmjs.com/package/lunr-languages and  https://www.npmjs.com/package/lunr as dependencies to  get dependabot to track for changes.

Comment (infotexture) [reply package.json]: Added in abcd6a8.

**Pull request merged**